src/pyhf/infer/calculators.py

```diff
@@ -137,6 +151,22 @@ def __init__(self, data, pdf, init_pars=None, par_bounds=None, qtilde=False):
         self.pdf = pdf
         self.init_pars = init_pars or pdf.config.suggested_init()
         self.par_bounds = par_bounds or pdf.config.suggested_bounds()
+
+        # get fixed vals from the model
+        model_fixed_vals = [
+            (index, init)
+            for index, (init, is_fixed) in enumerate(
+                zip(init_pars, fixed_params or pdf.config.suggested_fixed())
+            )
+            if is_fixed
+        ]
+        # add user-defined ones at the end
+        fixed_vals = model_fixed_vals + (fixed_vals or [])
+
+        # de-dupe and use last-appended result for each index
+        fixed_vals = list(dict(fixed_vals).items())
+
+        self.fixed_vals = fixed_vals
```

Comment:
```suggestion
        # de-dupe and use last-appended result for each index
        self.fixed_vals = list(dict(fixed_vals).items())
```

Maybe just save a few lines.